src/psbt/finalizer.rs

```diff
@@ -274,20 +274,12 @@ pub fn finalize_helper<C: secp256k1::Verification>(
     for (n, input) in psbt.inputs.iter().enumerate() {
         let target = input.sighash_type.unwrap_or(bitcoin::EcdsaSigHashType::All);
         for (key, ecdsa_sig) in &input.partial_sigs {
-            let flag = bitcoin::EcdsaSigHashType::from_u32_standard(ecdsa_sig.hash_ty as u32)
-                .map_err(|_| {
-                    super::Error::InputError(
-                        InputError::Interpreter(interpreter::Error::NonStandardSigHash(
-                            ecdsa_sig.to_vec(),
-                        )),
-                        n,
-                    )
-                })?;
-            if target != flag {
+            if target != ecdsa_sig.hash_ty {
+                // hash_ty is guaranteed to follow standardness rules
```

Comment:
In 7f4adf5e96a65adf78e8606be7ac86f3d272dad4:

I think this comment can be dropped -- standardness isn't actually relevant to this check, which is just that the signature hashtype matches the one required by the PSBT.

Also, I don't think that standardness is actually guaranteed here. See this code: https://github.com/rust-bitcoin/rust-bitcoin/pull/669#discussion_r780689066

Reply:
I think that is probably a bug upstream because we will risk generating wrong signatures. Anything that is consensus valid sighashtype byte (0x01, 0x02, x03, 0x81, 0x82, and 0x83) is mapped to sighash all. Consider a case with 0x05 sighash byte, while signing, we sighash_u32 as `0x05u32`, but while remapping it, we check it as `0x01` because the information that sighash_all from `0x05` instead of `0x01` is lost.

Reply:
oo good catch!

I think the bug is actually that we're losing information, rather than that we allow non-standard sighash types in PSBT. I can see how this happened, since this recent PSBT thing is the first time that we've ever deliberately supported non-standard sighash types..

Reply:
I opened https://github.com/rust-bitcoin/rust-bitcoin/issues/777 about this.

For now I still think we should remove this comment, because standardness is irrelevant, although you could add a comment linking to that rust-bitcoin issue.